File: Euler/java/euler4.java
package Euler.java;

public class euler4 {
    public String revert(String name)
    {
        StringBuffer sb = new StringBuffer(name);
        String reversedStr=sb.reverse().toString();
        return reversedStr;
    }
    static public void main(String[] args)
    {
        euler4 euler = new euler4();
        String keep;
        int ans = 0;
        for (int i=317;i<1000;i++)
        {
            for (int j=317;j<1000;j++)
            {
                keep=Integer.toString(i*j);
                String front=keep.substring(0,3);
                keep=euler.revert(keep);
                String end=keep.substring(0,3);
                if (front.equals(end))
                {
                    if (i*j>ans)
                    {
                        ans=i*j;
                    }
                }
            }
        }
        System.out.println(ans);
    }
}

File: Euler/python/euler4.py
ans=0
for i in range(999):
    for j in range(999):
        x=i*j
        k=str(x)
        rk=k[::-1]
        if len(k)==6:
            if k[0:3]==rk[0:3]:
                if x>ans:
                    ans=x
print(ans)